lib/THC/THCCachingAllocator.cpp

```diff
@@ -158,12 +158,12 @@ struct THCCachingAllocator
     allocated_blocks.erase(it);
 
```

Comment:
The only way to simplify it would be to introduce counter of free memory and update it on each allocation/deallocation.
That would be a sensible approach as well. I was shooting for a least intrusive method that does not add any complexity (and locks) to the main functionality.